src/Integration/Helpers/AbsoluteFilePathLocator.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.Helpers
+{
+    [Export(typeof(IAbsoluteFilePathLocator))]
+    internal class AbsoluteFilePathLocator : IAbsoluteFilePathLocator
+    {
+        private readonly IVsUIShellOpenDocument vsUiShellOpenDocument;
+
+        [ImportingConstructor]
+        public AbsoluteFilePathLocator([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            vsUiShellOpenDocument = serviceProvider.GetService<SVsUIShellOpenDocument, IVsUIShellOpenDocument>();
+        }
+
+        public string Locate(string relativeFilePath)
+        {
+            if (relativeFilePath == null)
+            {
+                throw new ArgumentNullException(nameof(relativeFilePath));
+            }
+
+            relativeFilePath = relativeFilePath.TrimStart('\\');
```

Comment:
Do we need to support forward slashes as well, or does the API handle them? I'm not sure what we'll get back from SonarQube for a C#/C++ file that is analysed on Linux machine.

Reply:
I'd expect the API to handle it, similarly to the way it does for issues, but we could verify it with SQ team when the hotspots api is finalized. 

https://github.com/SonarSource/sonarlint-visualstudio/blob/5fb393022542266ae2e8f93b362539ecfc127796/src/Integration/Suppression/SonarQubeIssuesProvider.cs#L128